### PR TITLE
Chitter

### PR DESCRIPTION
**Pull request by kevinlanzon**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'data_mapper'
+require './app/data_mapper_setup'
+
+task :auto_upgrade
```
Comment (spike01) [inline Rakefile]: Missing `do`

File: app/controllers/peeps.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Chitter < Sinatra::Base
+
+  get '/peeps/new' do
+    erb :"peeps/new"
+  end
+
+  post '/peeps' do
+    @peep = Peep.create(:post => params[:post],
+                     :user_name => params[:user_name])
```
Comment (spike01) [inline app/controllers/peeps.rb]: Indentation!

File: app/helpers/application.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Chitter < Sinatra::Base
+
+  helpers do
+
+    def current_user
+      @current_user ||=User.get(session[:user_id]) if session[:user_id]
```
Comment (spike01) [inline app/helpers/application.rb]: Space after the `||=`

File: app/views/flash.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<center>
```
Comment (spike01) [inline app/views/flash.erb]: Centering should happen in your CSS, not your tags

File: spec/peep_spec.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      peep = Peep.first
+      expect(peep.user_name).to eq('klanzon')
+      expect(peep.post).to eq('this is my first peep!')
+      peep.destroy
+      expect(Peep.count).to eq(0)
+    end
+
+  end
+
+end
```
Comment (spike01) [inline spec/peep_spec.rb]: Yes, this test was in the bookmark manager tutorial to show you how Datamapper works, but no, this should not be in your challenge

File: app/views/index.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  <% @peeps.each do |peep| %>
+    <% @peeps %>
+    <li><%= peep.post%></li>
+  <% end %>
+</ul>
+<% end %>
+<div id="new-peep">
+  <form action="/peeps" method="post">
+    <li><p>Create Peep: <input type="text" size="50" name="post"></p></li>
+    <li><p>Username: <input type="text" name="user_name"></p></li>
```
Comment (spike01) [inline app/views/index.erb]: Would this not be the user that is currently logged in?

File: spec/features/user_management_spec.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+include SessionHelpers
+
+feature "User signs up" do
+
+  scenario "when first visiting the homepage" do
+    expect{ sign_up }.to change(User, :count).by(1)
+    expect(page).to have_content("Welcome, chitty1")
+    expect(User.first.name).to eq("John Smith")
+    expect(User.first.username).to eq("chitty1")
+    expect(User.first.email).to eq("test@test.com")
```
Comment (spike01) [inline spec/features/user_management_spec.rb]: Don't test your database objects directly in a feature test

**Comment by spike01** [comment]
I've not left many comments on this because a.) I couldn't get any tests to run or even the server to start up without a lot tweaking and b.) most of what I'll say will be covered when we do a session with your array. The main things to think about are: 

- cleaning up your tests
- testing at the right layer
- not having multiple expectations
- implementing the features that were required on the spec

:construction:

File: app/views/index.erb (hunk 6 above)
Comment (tansaku) [inline app/views/index.erb]: kevin; sorry I didn't immediately follow up on this - end of the day and all that - looking over it I think @spike01 's point is that you shouldn't ask the user to specify who creates the peep - you should set the user who is currently logged in as the user associated with the peep automatically, and BTW, do ask follow up questions within the pull request - it's the developer's way!

File: Rakefile [hunk 8]
```diff
@@ -0,0 +1,12 @@
+require 'data_mapper'
+require './app/data_mapper_setup'
```
Comment (spike01) [inline Rakefile]: You'll need to require the models as well for this to work, you could also include the server

**Pull request closed without merging**